backend/src/main/scala/com/softwaremill/adopttapir/template/ProjectGenerator.scala

```diff
@@ -0,0 +1,212 @@
+            helloServerStub.body,
+            booksServerStub.body,
+            starterDetails.scalaVersion
+          )
+      } else {
+        val unwrapper = EndpointsSpecView.Unwrapper.prepareUnwrapper(starterDetails.serverEffect, starterDetails.scalaVersion)
+        txt
+          .EndpointsSpec(
+            starterDetails,
+            toSortedList(helloServerStub.imports ++ booksServerStub.imports ++ unwrapper.imports),
+            helloServerStub.body,
+            unwrapper.body,
+            booksServerStub.body,
+            starterDetails.scalaVersion
+          )
+      }
+
+    GeneratedFile(
+      pathUnderPackage("src/test/scala", groupId, "EndpointsSpec.scala"),
+      fileContent.toString
+    )
+  }
+
+  import CommonObjectTemplate.scalafmtConfigPath
+  private val scalafmtConf: ScalaVersion => GeneratedFile = dialectVersion =>
+    GeneratedFile(scalafmtConfigPath, txt.scalafmt(TemplateDependencyInfo.scalafmtVersion, dialectVersion).toString())
+
+  private def pathUnderPackage(prefixDir: String, groupId: String, fileName: String): String =
+    prefixDir + "/" + groupId.split('.').mkString("/") + "/" + fileName
+
+  private def toSortedList(set: Set[Import]): List[Import] = set.toList.sortBy(_.fullName)
+}
+
+object SbtProjectTemplate extends ProjectTemplate {
+  override def generate(starterDetails: StarterDetails): List[GeneratedFile] =
+    super.generate(starterDetails) ::: List(getBuildSbt(starterDetails), buildProperties, pluginsSbt, sbtx, readme)
+
+  lazy val sbtxFile = "sbtx"
+
+  private def getBuildSbt(starterDetails: StarterDetails): GeneratedFile = {
+    val content = txt
+      .sbtBuild(
+        starterDetails.projectName,
+        starterDetails.groupId,
+        starterDetails.scalaVersion.value,
+        TemplateDependencyInfo.tapirVersion,
+        (BuildSbtView.getAllDependencies _).andThen(BuildSbtView.format)(starterDetails),
+        starterDetails.serverEffect == ZIOEffect
+      )
+      .toString()
+
+    GeneratedFile("build.sbt", content)
+  }
+
+  private lazy val buildProperties: GeneratedFile = GeneratedFile(
+    "project/build.properties",
+    txt.buildProperties(TemplateDependencyInfo.sbtVersion).toString()
+  )
+
+  private lazy val pluginsSbt: GeneratedFile = GeneratedFile("project/plugins.sbt", CommonObjectTemplate.templateResource("plugins.sbt"))
+
+  private lazy val sbtx: GeneratedFile =
+    GeneratedFile(sbtxFile, CommonObjectTemplate.templateResource(sbtxFile))
+
+  private lazy val readme: GeneratedFile =
+    GeneratedFile(CommonObjectTemplate.readMePath, CommonObjectTemplate.templateResource("README_sbt.md"))
+}
+
+object CommonObjectTemplate {
+  def templateResource(fileName: String): String = Resource.getAsString(s"template/$fileName")
+
+  val scalafmtConfigPath = ".scalafmt.conf"
+
+  val readMePath: String = "README.md"
+
+  implicit class StarterDetailsWithLegalizedGroupId(starterDetails: StarterDetails) {
```

Comment:
maybe we should simply change the requirements for the package name so that invalid group ids are rejected?